### PR TITLE
Add styling to splash page, account setup, and submission creation views

### PR DESCRIPTION
**Pull request by MarshallMcCoy**
Simulator Screen Shot - iPhone 11 - 2020-04-30 at 10 42 27
Simulator Screen Shot - iPhone 11 - 2020-04-30 at 10 42 41

Simulator Screen Shot - iPhone 11 - 2020-04-30 at 10 56 28
I was unable to remove the extra padding that is being applied to the left of the table cell. @reddy88 do you have any ideas?

### CODE REVIEW

File: QStreak/View Controllers/Account Setup/AccountSetupViewController.swift [hunk 1]
```diff
@@ -14,11 +14,9 @@ class AccountSetupViewController: UIViewController {
 
     @IBOutlet private weak var zipCodeTextField: UITextField!
 
-    @IBOutlet private weak var ageTextField: UITextField!
+    @IBOutlet private weak var continueButton: QButton!
```
Comment (imreddy88) [inline QStreak/View Controllers/Account Setup/AccountSetupViewController.swift]: Can you subclass `QButton` and do the styling there? That way if designs change we can make any changes in one location.

File: QStreak/View Controllers/Account Setup/AccountSetupViewController.swift [hunk 2]
```diff
@@ -14,11 +14,9 @@ class AccountSetupViewController: UIViewController {
...
 
-    @IBOutlet private weak var householdSizeTextField: UITextField!
-
-    @IBOutlet private weak var continueButton: UIButton!
+    @IBOutlet weak var userNameTextField: QTextField!
```
Comment (imreddy88) [inline QStreak/View Controllers/Account Setup/AccountSetupViewController.swift]: Ditto: Subclass `QTextField`.

File: QStreak/View Controllers/Account Setup/WelcomeViewController.swift [hunk 3]
```diff
@@ -12,13 +12,12 @@ class WelcomeViewController: UIViewController {
 
     // MARK: - IBOutlets
 
-    @IBOutlet private weak var getStartedButton: UIButton!
+    @IBOutlet private weak var getStartedButton: QButton!
```
Comment (imreddy88) [inline QStreak/View Controllers/Account Setup/WelcomeViewController.swift:15]: Subclass

File: QStreak/View Controllers/Add Record/AddRecordViewController.swift [hunk 4]
```diff
@@ -9,16 +9,56 @@
 import UIKit
 
 class AddRecordViewController: UIViewController {
+    // TODO: move these to a constants file?
+    let teal = UIColor(red: 0.55, green: 0.96, blue: 0.82, alpha: 0.30)
```
Comment (imreddy88) [inline QStreak/View Controllers/Add Record/AddRecordViewController.swift]: Is this used anywhere?

File: QStreak/View Controllers/Add Record/AddRecordViewController.swift [hunk 5]
```diff
@@ -9,16 +9,56 @@
...
+    let grey = UIColor(red: 0.95, green: 0.96, blue: 0.97, alpha: 1.00)
 
     // MARK: - IBOutlets
```
Comment (imreddy88) [inline QStreak/View Controllers/Add Record/AddRecordViewController.swift:15]: Should all be private.

File: QStreak/View Controllers/Add Record/AddRecordViewController.swift [hunk 6]
```diff
@@ -9,16 +9,56 @@
...
+
+    // MARK: - IBActions
+
+    @IBAction func saveButtonTapped(_ sender: Any) {
+        viewModel.saveButtonTapped(date: Date(), contactCountString: contactCountTextField.text, selectedIndexPaths: tableView.indexPathsForSelectedRows)
+    }
+
+    @IBAction func decrementNumberOfPeopleButtonTapped(_ sender: Any) {
+        if let fieldValue = self.contactCountTextField.text {
+            let newNum = (Int(fieldValue) ?? 0) - 1
```
Comment (imreddy88) [inline QStreak/View Controllers/Add Record/AddRecordViewController.swift]: Move this logic into the `viewModel`.

File: QStreak/View Controllers/Add Record/AddRecordViewController.swift [hunk 7]
```diff
@@ -9,16 +9,56 @@
...
+
+            setDecrementNumberOfPeopleButtonStyles(newNumber: newNum)
+
+            self.contactCountTextField.text = "\(newNum)"
+        }
+    }
+
+    @IBAction func incrementNumberOfPeopleButtonTapped(_ sender: Any) {
+        if let fieldText = self.contactCountTextField.text {
```
Comment (imreddy88) [inline QStreak/View Controllers/Add Record/AddRecordViewController.swift:44]: DItto: Move to `viewModel`

File: QStreak/View Controllers/Add Record/AddRecordViewController.swift [hunk 8]
```diff
@@ -9,16 +9,56 @@
...
+
+            setDecrementNumberOfPeopleButtonStyles(newNumber: newNum)
 
-    @IBOutlet private weak var saveButton: UIButton!
+            self.contactCountTextField.text = "\(newNum)"
+        }
+    }
+
+    @IBAction func contactCountTextFieldEditingChanged(_ sender: Any) {
+        if let fieldText = self.contactCountTextField.text {
```
Comment (imreddy88) [inline QStreak/View Controllers/Add Record/AddRecordViewController.swift:54]: `viewModel`

File: QStreak/View Controllers/Add Record/AddRecordViewController.swift [hunk 9]
```diff
@@ -64,19 +114,23 @@ extension AddRecordViewController: UITableViewDataSource, UITableViewDelegate {
     }
 
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: "categoryCell", for: indexPath)
+        let cell = tableView.dequeueReusableCell(withIdentifier: "activityCell", for: indexPath)
         cell.textLabel?.text = viewModel.categories[indexPath.row].name
+
+        let iconName = tableView.indexPathsForSelectedRows?.contains(indexPath) ?? false ? "checkmark.square" : "square"
```
Comment (imreddy88) [inline QStreak/View Controllers/Add Record/AddRecordViewController.swift]: I think you can use `cell.isSelected`.

File: QStreak/View Controllers/Add Record/AddRecordViewController.swift [hunk 10]
```diff
@@ -9,16 +9,56 @@
...
-    @IBOutlet private weak var contactCountTextField: UITextField!
+    @IBOutlet private weak var tableView: UITableView!
 
-    @IBOutlet private weak var dateTextField: UITextField!
+    @IBOutlet weak var submissionDateLabel: UILabel!
 
-    @IBOutlet private weak var tableView: UITableView!
+    @IBOutlet weak var contactCountTextField: QTextField!
+
+    @IBOutlet weak var decrementNumberOfPeopleButton: UIButton!
```
Comment (imreddy88) [inline QStreak/View Controllers/Add Record/AddRecordViewController.swift]: Can you subclass `UIButton` and do your styling there? That way we have one place to make changes for this type of button.

**Review by imreddy88 — CHANGES_REQUESTED** [review]

**Comment by imreddy88** [comment]
Sry missed your comment about the table cell. It looks like the cell doesn't have any constraints.

File: QStreak/View Controllers/Add Record/AddRecordViewController.swift (hunk 9 above)
Comment (MarshallMcCoy) [reply QStreak/View Controllers/Add Record/AddRecordViewController.swift]: I originally tried using `isSelected` and it didn't seem to have the desired result (i.e., is selected was returning false at the point in time this was executed).

**Pull request merged**